src/main/code/de/tuchemnitz/se/exercise/core/graphics/zoommaps/ZoomMapsView.kt

```diff
@@ -3,60 +3,117 @@ package de.tuchemnitz.se.exercise.core.graphics.zoommaps
+import de.tuchemnitz.se.exercise.core.graphics.MainApp
+import de.tuchemnitz.se.exercise.core.graphics.system.ToolSelectionView
 import javafx.beans.property.SimpleBooleanProperty
 import javafx.beans.property.SimpleDoubleProperty
-import javafx.beans.property.SimpleObjectProperty
 import javafx.geometry.Point2D
 import javafx.geometry.Rectangle2D
 import javafx.scene.image.Image
 import javafx.scene.image.ImageView
 import javafx.scene.input.KeyCode
+import javafx.scene.input.KeyEvent
+import javafx.scene.layout.BorderPane
+import javafx.scene.layout.VBox
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 import tornadofx.View
-import tornadofx.borderpane
-import tornadofx.center
+import tornadofx.action
+import tornadofx.button
 import tornadofx.getValue
 import tornadofx.imageview
+import tornadofx.keyboard
 import tornadofx.setValue
 
+/*
+* This view is responsible for zooming in and out on a picture.
+*/
 class ZoomMapsView : View("Zoom Maps") {
-    private val clickLocationProperty: ObjectProperty<Point2D> = SimpleObjectProperty()
-    var clickLocation: Point2D by clickLocationProperty
-
+    /*
+    * @param root: unique root BorderPane for all Views
+    * @param contentBox: unique contentBox, where content is placed for every view
+    */
+    override val root: BorderPane by fxml(MainApp.MAIN_VIEW_TEMPLATE_PATH)
+    private val contentBox: VBox by fxid("content")
+
+    /*
+    * @param scaleProperty: injected property for the zoomFunction. This has to be edited in order to change
+    * the scale/speed of the zoom.
+    */
     private val scaleProperty = SimpleDoubleProperty(1.0)
     var scale by scaleProperty
 
+    /*
+    * @param configManager: the ConfigManager is being injected to save and
+    * get settings for the View (zoomSpeed, zoomButtons, etc.).
+    */
     private val configManager: ConfigManager by inject()
 
+    /*
+    * @param zoomEnabledProperty: states whether the initial zoom is activated or not (default is false).
+    */
     private val zoomEnabledProperty = SimpleBooleanProperty(false)
     var zoomEnabled by zoomEnabledProperty
 
+    /*
+    * Contains Logger, which is logging scroll and zoom Events.
+    */
     companion object {
         val logger: Logger = LoggerFactory.getLogger("ZoomMapsView Logger")
     }
 
-    override val root = borderpane {
-        val zoomKey = configManager.getZoomMapsConfig()?.zoomKey ?: KeyCode.E
-        setOnKeyPressed { e ->
-            logger.info("onKeyPressed: $e")
-            if (e.code == zoomKey) {
-                zoomEnabled = true
+    /*
+    * With the initialisation, the root of the fxml is being overwritten.
+    */
+    init {
+        with(root) {
+
+            /*
+             * @param zoomKey: the zoomKey is retrieved from the settings of the configManager,
+             * in case an error is occurring, it is being set to the char "E"
+             */
+            val zoomKey = configManager.getZoomMapsConfig()?.zoomKey ?: KeyCode.E
+
```

Comment:
ich meine alles was in der `with(root) { }` scope ist. wozu braucht man das?

Reply:
Um den Eventfilter auf root zu binden (oder eben woanders hin), und nicht nur auf die Contentbox vom tornadofxml. 
Hab "with root" jetzt allerdings raus genommen und alles auf die Contentbox gebindet. Der Unterschied is also im Grunde, ob der eventfilter auch auf Events im root object wartet. Muss er aber in diesem Fall gar nicht zwingend.